Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/Taa.azsl

```diff
@@ -99,11 +99,20 @@ float3 YCoCgToRgb(float3 yCoCg)
 // This function samples a 4x4 neighborhood around the uv. By taking advantage of bilinear filtering this can be
 // done with only 9 taps on the edges between pixels. The cost is further reduced by dropping the 4 diagonal
 // samples as their influence is negligible.
-float4 SampleCatmullRom5Tap(Texture2D<float4> texture, SamplerState linearSampler, float2 uv, float2 textureSize, float2 rcpTextureSize, float sharpness)
+float3 SampleCatmullRom5Tap(Texture2D<float4> texture, SamplerState linearSampler, float2 uv, float2 textureSize, float2 rcpTextureSize, float sharpness)
 {
     // Think of sample locations in the 4x4 neighborhood as having a top left coordinate of 0,0 and
     // a bottom right coordinate of 3,3.
 
+    // Use the local area green value as a measurement of luminance, and tweak the sharpness
+    // based on the difference between the max and min values. This will reduce the contribution
+    // of negative weights in high contrast areas and help prevent ringing artifacts.
+    float4 gatheredGreen = texture.GatherGreen(linearSampler, uv, int2(0, 0));
+    float minGreen = min(gatheredGreen.x, min(gatheredGreen.y, min(gatheredGreen.z, gatheredGreen.w)));
+    float maxGreen = max(gatheredGreen.x, max(gatheredGreen.y, max(gatheredGreen.z, gatheredGreen.w)));
+    float diff = maxGreen - minGreen;
+    sharpness /= diff + 1.0;
```

Comment:
Are you sure this operands is doing the following:
sharpness = sharpness / (diff + 1.0) 
and not: 
sharpness = sharpness / diff + 1.0

For clarity and safety I would change to:
sharpness /= (diff + 1.0f);

Reply:
hlsl follows the same operator precedence as c++, and all the assignment operators are very low precedence so they happen last (only thing lower precedence is a comma). I can add parenthesis though if that helps with clarity.